### ISSUE TITLE
[BUG] VolumeMounts not found: "datadog"

### ISSUE THREAD

**Issue by Ramilito** [DataDog/datadog-agent#16402]
**Agent Environment**

No agent gets created.

**Describe what happened:**
Pods not getting created on some nodes.
Checking the kubectl events shows this error:
```
Error creating: Pod "datadog-96z27" is invalid: [spec.initContainers[0].volumeMounts[2].name: Not found: "datadog", spec.initContainers[1].volumeMounts[5].name: Not found: "datadog"]
```
Manually adding the volumes to the helm chart for the datadog daemonset solves it.

**Describe what you expected:**
Pods to be created on all nodes

**Steps to reproduce the issue:**
Not sure how, seems to be happening when node is creating, deleting the datadog daemonset and recreating it seems to help.

**Additional environment details (Operating System, Cloud provider, etc):**
Cloud provider: AWS
Scheduler: Karpenter
Deployment: helm chart version: 3.23.0

**Comment by clamoriniere** [comment]
Hi @Ramilito 

To ease issue tracking, could you please open an issue on the https://github.com/DataDog/helm-charts repository where the `datadog` helm chart is located.

Also could you please provide in the new issue the `values.yaml` file  and the `helm` command arguments used to install the agent. 🙇

**Comment by cam3ron2** [comment]
Seeing the same thing with charts > 3.10.6
Chart: 3.25.1
Provider: AKS
K8s Version: 1.25.5

**Comment by cam3ron2** [comment]
Issue created on the charts repo:
https://github.com/DataDog/helm-charts/issues/986

**Comment by cam3ron2** [comment]
I just noticed that since upgrading the chart I have other pods failing because of PSP violations. Example:

```shell
Error creating: pods "gatekeeper-update-crds-hook-kd2lv" is forbidden: violates PodSecurity "restricted:v1.24": restricted volume types (volume "datadog" uses restricted volume type "hostPath")
```
It suspect that the datadog-cluster-agent 7.43.1 (bundled in this chart) is mutating that volume into all created pods, including the datadog-agent which is shooting the init containers in the foot.

**Comment by kubaracek** [comment]
Just ran into this. Datadog is blocking our app pods from being deployed

**Comment by CharlyF** [comment]
I left this note in the associated issue, while we investigate I hope this can unblock you @kubaracek https://github.com/DataDog/helm-charts/issues/986#issuecomment-1498296454

**Comment by cam3ron2** [comment]
Adding the following label to the gatekeeper job pod fixed that issue:

podLabels.admission.datadoghq.com/enabled=false

One thing that I find confusing is that when using chart 3.10.6 and older we did not see this issue, despite having ALWAYS had clusterAgent.admissionController.mutateUnlabelled set to true in our values.yaml.

@CharlyF is the issue there that the volme was previously readOnly?

**Comment by L3n41c** [comment]
I’ve understood the issue and proposed a fix in the cluster-agent: #16567. It should hopefully be part of 7.45.0 release.

What triggers the bug based on a specific version of the Helm chart is the modification of the default mode to `socket`.
That’s why explicitly specifying `mode: hostip` as @CharlyF suggested in [his comment](https://github.com/DataDog/helm-charts/issues/986#issuecomment-1498296454) should restore the previous behavior with the latest version of the chart.
